text/0497-glue-l2-construct.md

```diff
@@ -0,0 +1,418 @@
+certain types of data stores.
+
+* **Secrets Management
+    **User needs to specify JDBC connection credentials in Secrets Manager and
+    provide the Secrets Manager Key name as a property to the Job connection
+    property.
+
+* **Networking - CDK determines the best fit subnet for Glue Connection
+configuration
+    **The current glue-alpha-module requires the developer to specify the
+    subnet of the Connection when it’s defined. This L2 RFC will make the
+    best choice selection for subnet by default by using the data source
+    provided during Job provisioning, traverse the source’s existing networking
+    configuration, and determine the best subnet to provide to the Glue Job
+    parameters to allow the Job to access the data source. The developer can
+    override this subnet parameter, but no longer has to provide it directly.
+
+## Public FAQ
+
+### What are we launching today?
+
+We’re launching new features to an AWS CDK Glue L2 Construct to provide
+best-practice defaults and convenience methods to create Glue Jobs, Connections,
+Triggers, Workflows, and the underlying permissions and configuration.
+
+### Why should I use this Construct?
+
+Developers should use this Construct to reduce the amount of boilerplate
+code and complexity each individual has to navigate, and make it easier to
+create best-practice Glue resources.
+
+### What’s not in scope?
+
+Glue Crawlers and other resources that are now managed by the AWS LakeFormation
+team are not in scope for this effort. Developers should use existing methods
+to create these resources, and the new Glue L2 construct assumes they already
+exist as inputs. While best practice is for application and infrastructure code
+to be as close as possible for teams using fully-implemented DevOps mechanisms,
+in practice these ETL scripts will likely be managed by a data science team who
+know Python or Scala and don’t necessarily own or manage their own
+infrastructure deployments. We want to meet developers where they are, and not
+assume that all of the code resides in the same repository, Developers can
+automate this themselves via the CDK, however, if they do own both.
+
+Uploading Job scripts to S3 buckets is also not in scope for this effort.
+
+Validating Glue version and feature use per AWS region at synth time is also
+not in scope. AWS’ intention is for all features to eventually be propagated to
+all Global regions, so the complexity involved in creating and updating region-
+specific configuration to match shifting feature sets does not out-weigh the
+likelihood that a developer will use this construct to deploy resources to a
+region without a particular new feature to a region that doesn’t yet support
+it without researching or manually attempting to use that feature before
+developing it via IaC. The developer will, of course, still get feedback from
+the underlying Glue APIs as CloudFormation deploys the resources similar to the
+current CDK L1 Glue experience.
```

Comment:
While these limitations will not necessarily need to be validated in code, they will need to be documented in docstring.

Reply:
Noted.